scripts/restart-server

```diff
@@ -98,6 +98,14 @@ aux_services = list_supervisor_processes(["go-camo", "smokescreen"], only_runnin
 if aux_services:
     subprocess.check_call(["supervisorctl", "start", *aux_services])
 
+
+def restart_or_start(service: str) -> None:
+    our_verb = action
+    if our_verb == "restart" and len(list_supervisor_processes([service], only_running=True)) == 0:
```

Comment:
Should we avoid running `list_supervisor_processes` in a loop?  I guess I'm not sure how fast it is but it seems like it could be expensive...

Reply:
It's an RPC call, which is quite fast:
```
In [1]: from scripts.lib.supervisor import list_supervisor_processes

In [2]: %timeit list_supervisor_processes(['zulip-django'])
3.65 ms ± 105 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

I can optimize it by writing a cached function to call `list_supervisor_processes(only_running=True)`, cache the result, and look up into it -- but I think that would sacrifice readability some.

Reply:
Yeah, 3.65ms is fine -- I was worried about the 350ms that is a `supervisorctl status`.  

The scale of that difference makes me now wonder whether we should be replacing all of our `supervisorctl restart` shell operations with RPC.

Reply:
8e35cdb3dafd925c8988a7ddc2bd2a8efc88f88f talks about this a little.  Specifically, error handling for start/stop/restart is a little more involved, and using `supervisorctl` has some advantages there.